post: using Git with two users (#51)
cloudx-labs/posts: posts/navarroaxel/07.using-git-with-two-users.md

### Can we use the same key for all the users?

It depends. The source code hostings, like GitHub, associate the SSH key to a specific user of the platform to know which user is pushing code to a Git repository, check the access, etc. If your company doesn’t use Github, con can also use the same key for a personal email in Github and your company’s email in GitLab.

## Cloning a repository with a specific SSH key

The `git clone` command uses the SSH protocol to connect to the Git hosting, and the SSH program in your machine uses the default key (usually located at `~/.ssh/id_ed25519`) for authentication. You should use a custom SSH command to clone the repository using this:

 ```bash
git clone --config core.sshCommand="ssh -i path/to/private_ssh_key" git@github.com:orgname/repo.git
```

🧠 Remember that, if the key name ends with `.pub`, e.g. `id_orgname.pub`, then that is the public key. You should use the key file without this extension because that's your private key.

> If you should use another author's email, configure it with the following commands:

Reviewer: ```suggestion
If you want to use a different email address, configure it with the following commands:
```

```bash
git config user.name "John Doe"
git config user.email john@example.com
```

💡 Use the `--global` argument to set default settings for all the Git repositories in the machine. Otherwise the `git config` command only sets configuration values for the repository in the current directory.

## Changing the key to an already cloned repository

If you've cloned a Git repository but you need to use a new SSH key to push code, you can use this command to set a new SSH command:

```bash
git config core.sshCommand "ssh -i path/to/private_ssh_key"
```

## Extra SSH tips

> Maybe your company has their Git repositories is a custom port 🙈 (`22` is the default port for SSH), you need to clone these repositories with the following option:

Reviewer: ```suggestion
Maybe your company has their Git repositories in a custom port 🙈 (`22` is the default port for SSH), you need to clone these repositories with the following option:
```
